app/views/recipes/_form.html.erb

```diff
@@ -13,33 +10,47 @@
         </div>
       </div>
 
-      <div class="relative">
-        <div class="absolute inset-0 flex items-center" aria-hidden="true">
-          <div class="w-full border-t border-gray-300"></div>
-        </div>
-        <div class="relative flex justify-start">
-          <span class="bg-white pr-2 text-sm text-gray-500">Ingredients</span>
-        </div>
-      </div>
+      <div data-controller="template-manager">
 
-      <div id="ingredients" class="space-y-3" data-streams-target="container">
-        <% recipe.ingredients.each do |ingredient| %>
-          <%= render "recipes/ingredient_fields", ingredient: ingredient %>
-        <% end %>
-      </div>
+        <!-- Ingredients section-->
+        <div class="relative">
+          <div class="absolute inset-0 flex items-center" aria-hidden="true">
+            <div class="w-full border-t border-gray-300"></div>
+          </div>
+          <div class="relative flex justify-start">
+            <span class="bg-white pr-2 text-sm text-gray-500">Ingredients</span>
+          </div>
+        </div>
+        <!-- Ingredients template-->
+        <template data-template-manager-target="template">
+          <%= form.fields_for "ingredients_attributes", index: "replace_with_epoch" do |template_form| %>
+            <%= render "template_field", form: template_form %>
+          <% end %>
+        </template>
 
-      <div class="relative">
-        <div class="absolute inset-0 flex items-center" aria-hidden="true">
-          <div class="w-full border-t border-gray-300"></div>
+        <!-- Ingredients list-->
+        <div id="ingredients" class="space-y-3" data-template-manager-target="container">
+          <%= form.fields_for :ingredients do |ingredients_form| %>
```

Comment:
Using `form.field_for :ingredients` will take care of adding an id to any element in the ingredients collection for the recipe.